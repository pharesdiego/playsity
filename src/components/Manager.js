import React, { Component } from 'react';

// this component helps us manage state
class Manager extends Component {
  state = {
    ...this.props.initialState
  }

  updateState = (prop, value) => {
    this.setState({
      [prop]: value
    })
  }

  render() {
    return this.props.render(this.state, this.updateState);
  }
}

export default Manager;