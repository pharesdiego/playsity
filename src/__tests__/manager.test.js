import React from 'react';
import Manager from './../components/Manager';

const Mock = (props) => {
  return (
    <button {...props} />
  )
}

const MockWithManager = (props) => {
  return (
    <Manager initialState={{passing: 'false'}} render={(state, updateState) => (
      <Mock {...state} onClick={() => updateState('passing', 'true')} />
    )}/>
  )
}

let wrapper;

describe('>>> <Manager> <<<', () => {
  beforeEach(() => {
    wrapper = mount(<MockWithManager/>);
  });

  it('Must render', () => {
    expect(wrapper).toBeDefined();
  });

  it('Must update state with updateState function and send it to child(s)', () => {
    wrapper.find('button').simulate('click');
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('button').props().passing).toBe('true');
  });
});
