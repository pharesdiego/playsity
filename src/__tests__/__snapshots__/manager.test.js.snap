// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> <Manager> <<< Must update state with updateState function and send it to child(s) 1`] = `
<MockWithManager>
  <Manager
    initialState={
      Object {
        "passing": "false",
      }
    }
    render={[Function]}
  >
    <Mock
      onClick={[Function]}
      passing="true"
    >
      <button
        onClick={[Function]}
        passing="true"
      />
    </Mock>
  </Manager>
</MockWithManager>
`;
